cmd/agent/gui/systray/Sources/gui.swift

```diff
@@ -209,10 +215,17 @@ class AgentManager {
             NSLog(processInfo.stdOut)
             NSLog(processInfo.stdErr)
         }
+        
 
         checkStatusAndCall(command: command, timeout: serviceTimeout, callback: callback)
     }
 
+    static func checkCurrentInstallationMode() -> Bool {
+        let process = bashCall(command: "test -f " + userAgentPlistPath)
```

Comment:
question: will the `/bin/launchctl unload -w` remove the plist file entirely?

my attempts locally suggest that it will not remove the file so this will always return true on a user installation?

```
venv ❯ test -f ~/Library/LaunchAgents/com.datadoghq.agent.plist && echo "plist exists"
plist exists

venv ❯ launchctl unload -w ~/Library/LaunchAgents/com.datadoghq.agent.plist

venv ❯ test -f ~/Library/LaunchAgents/com.datadoghq.agent.plist && echo "plist exists"
plist exists
```

Reply:
possibly we would want to try `launchctl print-disabled gui/$UID` for user installation or `launchctl print-disabled system` for headless?

```
venv ❯ launchctl print-disabled gui/502 | grep com.datadoghq.agent
                "com.datadoghq.agent" => disabled
```

Reply:
nevermind, this is working fine as it isn't needed to toggle based on the launchctl status